### PR TITLE
Refactors some trauma code, fixes some trauma heals not working

### LINKED ISSUE #36076: Brain Surgery gives you irreversible damage (and no, it wasn't from a lobotomy)
[Round ID]: # 84387

[Reproduction]: # Have tinfoil hat on. Melt it off with acid. Get surgery to remove phobia. Person fucks up surgery giving me brain damage and making me unable to talk. Get surgery again to fix it, nothing happens. Take both mannitol and mutadone pills, still nothing fixes it. Ask admins about it, they say that i have 0 brain damage and they aheal me to see if that fixes it either, nothing.

### LINKED ISSUE #36301: Mannitol does not cure brain traumas
Issue reported from  Round ID: 84709 (/tg/Station Bagil [ENGLISH] [US-EAST] [100% FREE LAG])

Reporting client version: 511

I'm talking about the mild/special traumas, that should be curable with mannitol, not the ones that need surgery.
Gave myself three mild traumas, took about 400u of mannitol, still had all traumas afterwards. I'm guessing it doesn't work for special traumas either.

### PR DESCRIPTION
**Pull request by XDTM**
Fixes #36076
Fixes #36301

Traumas now handle all the gain parts on gain_trauma instead of New().
You can now create traumas an apply them later to a target, which makes temporary traumas possible by storing them in a var.

### CODE REVIEW

**Comment by ShizCalev** [comment]
conflict.

**Comment by ShizCalev** [comment]
conflict.

File: code/modules/mob/living/brain/brain_item.dm [hunk 1]
```diff
@@ -222,16 +222,20 @@
...
-		SSblackbox.record_feedback("tally", "traumas", 1, trauma_type)
-		traumas += new trauma_type(arglist(list(src, resilience) + arguments))
+		trauma = new(arglist(arguments))
 	else
-		SSblackbox.record_feedback("tally", "traumas", 1, trauma.type)
-		traumas += trauma
-		if(resilience)
-			trauma.resilience = resilience
+		if(trauma.brain) //we don't accept used traumas here
+			return
```
Comment (AnturK) [inline code/modules/mob/living/brain/brain_item.dm]: Should probably throw a warning here.

File: code/modules/clothing/head/misc_special.dm [hunk 2]
```diff
@@ -236,16 +236,24 @@
 	item_state = "foilhat"
 	armor = list("melee" = 0, "bullet" = 0, "laser" = -5,"energy" = 0, "bomb" = 0, "bio" = 0, "rad" = -5, "fire" = 0, "acid" = 0)
 	equip_delay_other = 140
+	var/datum/brain_trauma/mild/phobia/paranoia
 
 /obj/item/clothing/head/foilhat/equipped(mob/living/carbon/human/user, slot)
+	..()
 	if(slot == slot_head)
-		user.gain_trauma(/datum/brain_trauma/mild/phobia, FALSE, "conspiracies")
+		paranoia = new()
```
Comment (AnturK) [inline code/modules/clothing/head/misc_special.dm]: Shouldn't it check if it already exists ?

**Comment by XDTM** [comment]
@AnturK addressed the review; probably best to merge this quickly, apparently traumas aren't healing at all right now

**Pull request merged**